### ISSUE TITLE
Understand FreqAI models lifecycle

### ISSUE THREAD

**Issue by escanoro** [freqtrade/freqtrade#8409]
## Describe your environment

  * Operating system: Ubuntu 22.04
  * Python Version: _____ N/A
  * CCXT version: _____ N/A
  * Freqtrade Version: docker: freqtradeorg/freqtrade:stable (Current stable version as of 03 28 2023)
  
## Your question

Greetings,

This question is to bring more understanding and clarity about how the models in FreqAI relates to each other.
After reading the documentation once and second, i can say that i'm a little bit confused. Let me ask my question with maximum clarity:

- Models are generated once per epoch (Once per training period), we can set how many models to keep on disk

- If we set `"continual_learning": true`, Freqai will use the last model as a starting point for next model:
  -  Is starting with a **Memory** better than starting from scratch ? (naive thinking may say that having a memory is better, additional thinking may say that it depends on the memory, if the future situations are radically different from the past, then having a memory is not an advantage) --> Am i correct ?

 - Let's say i have trained models on past data, since the last model is used for inferencing, older models are useless unless for comparison and similar purposes --> Am i correct ?
 

- When Back testing with `--freqaimodel`: Since the last model is used for inference, i assume that back testing results are only coming from the last model. If i want to Backtest using an older model, how can i achieve that? There is no way in the command lines to specify a specific model (or version)

- In **theory**, do a greater **"train_period_days"** improves the chances to obtain better models ? I mean training on 100 days of data is better then 1 day (In theory). In the other hand, such models cannot adapt to environment changes (market change in our case) --> Am i correct ?

Thanks to the Team, great framework!
Regards,
Escanoro

**Comment by robcaulk** [comment]
Hello,

Great questions, thanks for the curiosity. I appreciate that you brought your thoughts with the questions. 

>Models are generated once per epoch (Once per training period), we can set how many models to keep on disk

Correction: epoch is not the proper terminology. A new model is generated as often as you tell it to generate a new model with `live_retrain_hours` in dry/live, or with `backtest_period_days` in backtesting. Epoch, on the other hand, is reserved for identifying periods in training loops which are distinguished by the fact that all samples are seen once. Decision trees (e.g. XGBoost) do not have epochs. RL, PyTorchRegressor, and the CNN all use these training loops with an epoch parameter.

> Is starting with a Memory better than starting from scratch ? (naive thinking may say that having a memory is better, additional thinking may say that it depends on the memory, if the future situations are radically different from the past, then having a memory is not an advantage) --> Am i correct ?

You are thinking about the pros and cons correctly. I (and FreqAI) cannot make these choices for you. Maybe it is helpful to consider the limit cases, if you have extreme chaotic data that is repeating few patterns with time, and you have very small models that are quick to train - then it may not be beneficial to keep memory. If you have patterns repeating alot in your data, and you have huge models that take a long time to train - then perhaps it makes sense to start from the end state of the previous model.

>Let's say i have trained models on past data, since the last model is used for inferencing, older models are useless unless for comparison and similar purposes --> Am i correct ?

Correct. We use them to analyze the temporal evolution of [feature importance](https://emergentmethods.ai/experiments.html#ongoing) (purple plot at the bottom of the page), but you have no real need for historical models if you are not interested in processing their data. 

>When Back testing with --freqaimodel: Since the last model is used for inference, i assume that back testing results are only coming from the last model. If i want to Backtest using an older model, how can i achieve that? There is no way in the command lines to specify a specific model (or version)

You are mistaken. FreqAI is automating real-time adaptive modeling for you. That means it is retraining at some user defined frequency. Backtesting simulates this behavior by sliding a train window through your backtest period, retraining and making predictions all along the way. Finally aggregating all predictions and giving you the result. Please have a look at our doc where we have already explained this https://www.freqtrade.io/en/stable/freqai-running/#deciding-the-size-of-the-sliding-training-window-and-backtesting-duration.

>In theory, do a greater "train_period_days" improves the chances to obtain better models ? I mean training on 100 days of data is better then 1 day (In theory). In the other hand, such models cannot adapt to environment changes (market change in our case) --> Am i correct ?

This is an open research question - and one that FreqAI is specifically designed to help *you* test. You seem to have identified the two sides of the coin correctly, but I'd like to elaborate on this point. Neither I, nor FreqAI can make this choice for you. It depends on the periodicity of the features comprising your parameter space. Your parameter space is defined by the features you engineered. Maybe it is helpful to consider the limit cases again. If your prediction horizon is 7 days out, then you would need at *least* a few periods of 7 days to capture that periodicity in your parameter space so that the model could identify the pattern and make a legitimate prediction. Then you have this point that you mention of quickness of adapting to environment changes. 

Cheers,

Robert
